### PR TITLE
fix(coinmarket): design bugs & improvements

### LINKED ISSUE #13679: Coinmarket - redesign 1
image

### LINKED ISSUE #13719: Coinmarket - redesign design issues
## Design bugs
1. Overlap select and modal
image
2. Not displaying the received amount in the Buy detail transaction and partner logo with the name
image
3. The Receive method in error state is just blank, let’s add ‘No methods available'. [Figma link](https://www.figma.com/design/xSW1knAjH25SfOmqSdNioY/Trade?node-id=7651-16248&t=dxYKbMUF8mkxOiU3-1)
image
4.  The space between the chips and the input is not correct, it should be 8px.
image
5. Remove button title (https://github.com/trezor/trezor-suite/issues/13741)
image
6. Sell `All` works only on double-click
7. Fiat change in the Exchange section doesn't update fee levels and `Compare offers` is stuck.
8. Receive address design bug (https://github.com/trezor/trezor-suite/issues/13679)
9. Exchange badly selected cryptocurrency (BE problem)
10. https://github.com/trezor/trezor-suite/issues/13740
11. https://github.com/trezor/trezor-suite/issues/13739

### LINKED ISSUE #13739: Coinmarket - redesign Switching to fiat input in Sell doesn't switch it for the first time
**Describe the bug**
When I try to switch to fiat for the first time, it doesn't really happen and there is a forever loading in the "Receive method" part.

**Steps to reproduce:**
1. Go to your BTC account
2. Go to "Trade" section
3. Select "Sell" feature
4. Switch to EUR from BTC input
5. Observe that it doesn't switch and there's forever loading in the "Receive method" part below

**Info:**
 - Suite version: desktop 24.8.1 (b4f8ad7e864cf72f3965f41e541e280bfa906a7d)
 - Browser: Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) TrezorSuite/24.8.1 Chrome/118.0.5993.159 Electron/27.3.8 Safari/537.36
 - OS: Linux x86_64
 - Screen: 1920x1080
 - Device: Trezor T3T1 2.8.1 regular (revision 632b9561559b7ab6824bb7eeac072874e07b7b82)
 - Transport: BridgeTransport 2.0.33

**Expected result:**
Input from BTC to fiat switches correctly

**Video:**
https://youtu.be/vF8R6ucRgzE

### LINKED ISSUE #13740: Coinmarket - redesign When switching to fiat input in Sell I get an error
**Describe the bug**
When I switch to fiat input in the Sell section, I get almost infinite loading and a error that shouldn't be there since I didn't even input anything in the field.

**Steps to reproduce:**
1. Start Suite
2. Go to the "Trade" section and select the "Sell" feature
3. Try to switch from BTC to fiat input

**Info:**
 - Suite version: desktop 24.8.1 (b4f8ad7e864cf72f3965f41e541e280bfa906a7d)
 - Browser: Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) TrezorSuite/24.8.1 Chrome/118.0.5993.159 Electron/27.3.8 Safari/537.36
 - OS: Linux x86_64
 - Screen: 1920x1080
 - Device: Trezor T3T1 2.8.1 regular (revision 632b9561559b7ab6824bb7eeac072874e07b7b82)
 - Transport: BridgeTransport 2.0.33

**Expected result:**
There is no error with decimals since I didn't input anything.

**Screenshot:**
image

### LINKED ISSUE #13741: Coinmarket - redesign Missing tooltip translation
**Describe the bug**
There's a missing translation for tooltip in the Trade section of Suite on the "Last transactions" button (on hover).

**Steps to reproduce:**
1. Go to "Trade" section
2. Hover over the "Last transactions" button
3. Observe the missing translation on the tooltip

**Info:**
 - Suite version: desktop 24.8.1 (b4f8ad7e864cf72f3965f41e541e280bfa906a7d)
 - Browser: Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) TrezorSuite/24.8.1 Chrome/118.0.5993.159 Electron/27.3.8 Safari/537.36
 - OS: Linux x86_64
 - Screen: 1920x1080
 - Device: Trezor T3T1 2.8.1 regular (revision 632b9561559b7ab6824bb7eeac072874e07b7b82)
 - Transport: BridgeTransport 2.0.33

**Screenshots:**
image

### PR DESCRIPTION
**Pull request by adderpositive**
## Information 

Resolve: https://github.com/trezor/trezor-suite/issues/13719
Resolve: https://github.com/trezor/trezor-suite/issues/13679
Resolve: https://github.com/trezor/trezor-suite/issues/13741
Resolve: https://github.com/trezor/trezor-suite/issues/13739
Resolve: https://github.com/trezor/trezor-suite/issues/13740

## Design
https://www.figma.com/design/xSW1knAjH25SfOmqSdNioY/Trade?node-id=6032-34581&t=3moEF3un9o4cHg1A-1

### CODE REVIEW

**Review by LukasRada — APPROVED** [review]

File: packages/suite/src/views/wallet/coinmarket/common/CoinmarketSelectedOffer/CoinmarketSelectedOfferVerifyOptions.tsx [hunk 1]
```diff
@@ -53,104 +54,120 @@ const AccountType = styled.span`
     padding-left: 5px;
 `;
 
+const SelectWrapper = styled.div`
+    position: relative;
+    z-index: ${zIndices.navigationBar};
```
Comment (jvaclavik) [inline packages/suite/src/views/wallet/coinmarket/common/CoinmarketSelectedOffer/CoinmarketSelectedOfferVerifyOptions.tsx]: I'm not sure about this z-index. Does it have anything common with navigationBar?

File: packages/suite/src/views/wallet/coinmarket/common/CoinmarketSelectedOffer/CoinmarketSelectedOfferVerifyOptions.tsx [hunk 2]
```diff
@@ -53,104 +54,120 @@ const AccountType = styled.span`
...
+                                            •
+                                            <FiatWrapper>
+                                                <FiatValue
+                                                    amount={formattedBalance}
+                                                    symbol={symbol}
+                                                />
+                                            </FiatWrapper>
+                                        </Amount>
+                                    </AccountWrapper>
+                                </Option>
```
Comment (jvaclavik) [inline packages/suite/src/views/wallet/coinmarket/common/CoinmarketSelectedOffer/CoinmarketSelectedOfferVerifyOptions.tsx]: Could you please create new components from these switch cases? 🙏 I think it's easier to read it.

File: packages/suite/src/views/wallet/coinmarket/common/CoinmarketSelectedOffer/CoinmarketSelectedOfferVerifyOptions.tsx (hunk 1 above)
Comment (adderpositive) [reply packages/suite/src/views/wallet/coinmarket/common/CoinmarketSelectedOffer/CoinmarketSelectedOfferVerifyOptions.tsx]: No, it doesn't. Should I rather use a plain number?

Comment (jvaclavik) [reply packages/suite/src/views/wallet/coinmarket/common/CoinmarketSelectedOffer/CoinmarketSelectedOfferVerifyOptions.tsx]: In this case yes. Thanks 🙏

File: packages/suite/src/hooks/wallet/coinmarket/form/useCoinmarketVerifyAccount.tsx [hunk 3]
```diff
@@ -96,6 +96,7 @@ const useCoinmarketVerifyAccount = ({
     const isDebug = useSelector(selectIsDebugModeActive);
     const device = useSelector(selectDevice);
     const dispatch = useDispatch();
+    const [isMenuOpen, setIsMenuOpen] = useState<boolean | undefined>(undefined);
```
Comment (hstastna) [inline packages/suite/src/hooks/wallet/coinmarket/form/useCoinmarketVerifyAccount.tsx:99]: Maybe using just `boolean` values without `undefined` here would be sufficient. WDYT?

Comment (adderpositive) [reply packages/suite/src/hooks/wallet/coinmarket/form/useCoinmarketVerifyAccount.tsx:99]: It is used for one specific purpose, when `account.type === 'ADD_SUITE'` is when the modal will open and select stays opened after the action is completed in the modal. 

If the `isMenuOpen` is `undefined`, there is no need to manage the state for other cases.

**Review by MiroslavProchazka — APPROVED** [review]

**Review by tomasklim — DISMISSED** [review]
Probably it is not all related to this PR, but this is what I found checking this PR manually

1. 
Why is account name rerendering?

https://github.com/user-attachments/assets/6f18e793-09ba-4fda-a1eb-f0cd83d047c6

2.
Endless rerendering in compare offers

https://github.com/user-attachments/assets/4020ab90-e703-4fc4-9a37-13931e5321f3

3. 
My best offer is not best according to compare offers :( 

Screenshot 2024-08-12 at 22 49 34
Screenshot 2024-08-12 at 22 49 54

4. my best offer is not best at all
Screenshot 2024-08-12 at 22 50 34
Screenshot 2024-08-12 at 22 50 50

5. 
I am buying on Acc #2, why receive acc is by default ETH #1?

Screenshot 2024-08-12 at 22 51 10

6. By what logic you fill the amount there? I came with 100 usd and after 2 changes it is 150

https://github.com/user-attachments/assets/cef2ad36-d671-409c-b93e-42192a181c7d

7. Many inconsistent states by enterring amount in fiat. I get offer and I am fast enough I can get to payment gateway

https://github.com/user-attachments/assets/ae3c14d5-dc63-4a27-8b79-ad1660e1e2dc

8. You are showing hidden tokens 
Screenshot 2024-08-12 at 22 58 30

9. Beware of backup gas limit. You are setting 21000. The transaction will fail 100%.

10. Balances are not blurred 
Screenshot 2024-08-12 at 23 05 06

11. Sell form does not respect that I am on polygon network

https://github.com/user-attachments/assets/91b9a5b5-4025-4480-8161-2b00b7ee72b7

12. Can the selected value be the first one I see in select?

https://github.com/user-attachments/assets/6a0c696b-7f94-4afc-b9cb-5cdf21f59ae0
 
 13. Last transactions are not powered by Invity footer
Screenshot 2024-08-12 at 23 08 27

14. As I can get into really messy state with my form, I would expect `Clear` button as I have in staking and sending

**Comment by bosomt** [comment]
1. Im trying to sell my $408.72 but all button tries to sell 500+ Euro 
 Im not able to replicate it every time but after you change USD to EUR or something it can be broken

image

2. 1/3 ... all buttons are not available when you can enter fiat and not btc amount 
image

**Pull request merged**